src/fosslight_source/cli.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 LG Electronics Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+# import multiprocessing
+import warnings
+# import platform
+import getopt
+import logging
+# import yaml
+# import json
+import copy
+# from scancode import cli
+from datetime import datetime
+import fosslight_util.constant as constant
+from fosslight_util.set_log import init_log
+from fosslight_util.timer_thread import TimerThread
+# from ._parsing_scancode_file_item import parsing_file_item
+# from ._parsing_scancode_file_item import get_error_from_header
+from ._help import print_help_msg_source
+from ._license_matched import get_license_list_to_print
+from fosslight_util.output_format import check_output_format, write_output_file
+# from ._parsing_scanoss_file import parsing_scanResult # scanoss
+# from ._scan_item import ScanItem
+from .run_scancode import run_scan
+from .run_scanoss import run_scanoss_py
+
+logger = logging.getLogger(constant.LOGGER_NAME)
+warnings.filterwarnings("ignore", category=FutureWarning)
+_PKG_NAME = "fosslight_source"
+
+
+def main():
+    global logger
+    success = True
+    _result_log = {}
+
+    argv = sys.argv[1:]
+    path_to_scan = ""
+    write_json_file = False
+    output_file_name = ""
+    print_matched_text = False
+    format = ""
+    selected_scanner = ""
+
+    scanned_result = []
+    license_list = []
+    # result_list = []
+
+    try:
+        opts, args = getopt.getopt(argv, 'hmjs:p:o:f:')
+        for opt, arg in opts:
+            if opt == "-h":
+                print_help_msg_source()
+            elif opt == "-p":
+                path_to_scan = arg
+            elif opt == "-j":
+                write_json_file = True
+            elif opt == "-o":
+                output_file_name = arg
+            elif opt == "-m":
+                print_matched_text = True
+            elif opt == "-f":
+                format = arg
+            elif opt == "-s":
+                selected_scanner = arg.lower()
+    except Exception:
+        print_help_msg_source()
+
+    timer = TimerThread()
+    timer.setDaemon(True)
+    timer.start()
+
+    print("output_file_name : ", output_file_name)
+    start_time = datetime.now().strftime('%Y%m%d_%H%M%S')
+    success, msg, output_path, output_file, output_extension = check_output_format(output_file_name, format)
```

Comment:
This is called at create_report_file().

Reply:
It still is needed for log initialization.